### ISSUE TITLE
Expected behavior of `example_unsupervised.sh`

### ISSUE THREAD

**Issue by bkj** [williamleif/GraphSAGE#10]
Can you clarify the expected behavior of the unsupervised example script?  I'm getting something like:

```
...
Iter: 3150 train_loss= 0.00115 train_mrr= 0.16871 train_mrr_ema= 0.18267 val_loss= 0.00271 val_mrr= 0.18308 val_mrr_ema= 0.18308 time= 0.01506
Iter: 3200 train_loss= 0.00115 train_mrr= 0.17344 train_mrr_ema= 0.18270 val_loss= 0.00271 val_mrr= 0.18308 val_mrr_ema= 0.18308 time= 0.01506
Iter: 3250 train_loss= 0.00115 train_mrr= 0.15224 train_mrr_ema= 0.18271 val_loss= 0.00271 val_mrr= 0.18308 val_mrr_ema= 0.18308 time= 0.01504
Iter: 3300 train_loss= 0.00115 train_mrr= 0.17956 train_mrr_ema= 0.18197 val_loss= 0.00271 val_mrr= 0.18308 val_mrr_ema= 0.18308 time= 0.01504
Iter: 3350 train_loss= 0.00115 train_mrr= 0.20141 train_mrr_ema= 0.18416 val_loss= 0.00271 val_mrr= 0.18308 val_mrr_ema= 0.18308 time= 0.01506
Iter: 3400 train_loss= 0.00115 train_mrr= 0.20716 train_mrr_ema= 0.18435 val_loss= 0.00271 val_mrr= 0.18308 val_mrr_ema= 0.18308 time= 0.01507
Iter: 3450 train_loss= 0.00114 train_mrr= 0.19132 train_mrr_ema= 0.18373 val_loss= 0.00271 val_mrr= 0.18308 val_mrr_ema= 0.18308 time= 0.01506
Iter: 3500 train_loss= 0.00115 train_mrr= 0.18440 train_mrr_ema= 0.18375 val_loss= 0.00271 val_mrr= 0.18308 val_mrr_ema= 0.18308 time= 0.01508
Iter: 3550 train_loss= 0.00115 train_mrr= 0.19627 train_mrr_ema= 0.18545 val_loss= 0.00271 val_mrr= 0.18308 val_mrr_ema= 0.18308 time= 0.01512
Iter: 3600 train_loss= 0.00115 train_mrr= 0.17484 train_mrr_ema= 0.18656 val_loss= 0.00271 val_mrr= 0.18308 val_mrr_ema= 0.18308 time= 0.01513
Iter: 3650 train_loss= 0.00114 train_mrr= 0.19171 train_mrr_ema= 0.18581 val_loss= 0.00271 val_mrr= 0.18308 val_mrr_ema= 0.18308 time= 0.01513
Iter: 3700 train_loss= 0.00114 train_mrr= 0.19534 train_mrr_ema= 0.18489 val_loss= 0.00271 val_mrr= 0.18308 val_mrr_ema= 0.18308 time= 0.01514
Optimization Finished!
```

So the `train_loss` is decreasing, but the `val_loss` appears to be exactly the same.  Any thoughts?

Thanks
Ben

**Comment by RexYing** [comment]
Hi,
There's a flag (validate_iter) that sets the number of iterations between two validation runs. It's set to 5000 by default for efficiency reasons when running on large dataset. For this small dataset, you could set a smaller number.

**Comment by bkj** [comment]
Ah great, thanks -- on the example data, it looks like the default value is so high it never gets triggered.

Related question: I'm trying to reproduce the unsupervised results on the `reddit` and `ppi` datasets.  If I run:

```
python -m graphsage.unsupervised_train \
    --train_prefix ./data/reddit/reddit \
    --model graphsage_mean \
    --learning_rate 0.001

python ./eval_scripts/reddit_eval.py
```

should I get something close?

**Comment by williamleif** [comment]
Just pushed a change to have "--validate_iter 10" for the unsupervised example.

And yeah, those settings should get you in the ballpark. I recommend starting with a learning rate of 0.0001 though (this was usually optimal iirc), and the "big" models will probably perform a bit better (though with longer runtime).

**Comment by bkj** [comment]
Cool thanks -- will close
